build.rs

```diff
@@ -44,14 +44,21 @@ fn main() {
     //
     // On CI, we detect the actual FreeBSD version and match its ABI exactly,
     // running tests to ensure that the ABI is correct.
-    match which_freebsd() {
-        Some(10) if libc_ci => set_cfg("freebsd10"),
-        Some(11) if libc_ci => set_cfg("freebsd11"),
-        Some(12) if libc_ci || rustc_dep_of_std => set_cfg("freebsd12"),
```

Comment:
I just realized the previous logic for rustc_dep_of_std here wasn't what I thought it was...

The logic was: if the auto-detected version is 12 and `rustc_dep_of_std` is set, then use `freebsd12`. If the auto-detected version is anything else and `rustc_dep_of_std` is set, use `freebsd11`. If this was deliberate, I don't understand why, and it doesn't match the comments.

With my PR, the logic now is: if `rustc_dep_of_std` is set, don't even auto-detect, just use version 12. This matches what the comments already said before.